File: backend/src/routes/tournaments.ts
import { Router } from 'express';
import { TournamentController } from '../controllers/TournamentController';

const router = Router();
const tournamentController = new TournamentController();

// Get all tournaments
router.get('/', (req, res) => {
  tournamentController.getAllTournaments(req, res);
});

// Get tournament by ID
router.get('/:id', (req, res) => {
  tournamentController.getTournamentById(req, res);
});

// Create new tournament
router.post('/', (req, res) => {
  tournamentController.createTournament(req, res);
});

// Join tournament
router.post('/:id/join', (req, res) => {
  tournamentController.joinTournament(req, res);
});

// Get tournament participants
router.get('/:id/participants', (req, res) => {
  tournamentController.getTournamentParticipants(req, res);
});

export { router as tournamentRoutes };

File: backend/src/routes/tournaments.rs
use std::sync::Arc;
use axum::{routing::{get, post}, Router};
use crate::controllers::tournaments_controller as ctrl;
use crate::AppState;

pub fn router(state: Arc<AppState>) -> Router<Arc<AppState>> {
    Router::new()
        .route("/api/tournaments", get(ctrl::list).post(ctrl::create))
        .route("/api/tournaments/:id", get(ctrl::get_one))
        .route("/api/tournaments/:id/join", post(ctrl::join))
        .with_state(state)
}


